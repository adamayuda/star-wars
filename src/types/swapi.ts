export interface swapiPeople {
  name: string;
  height: string;
  mass: string;
  hair_color: string;
  skin_color: string;
  eye_color: string;
  birth_year: string;
  gender: string;
  homeworld: string;
  films: string[];
  species: string[];
  vehicles: string[];
  starships: string[];
  created: Date;
  edited: Date;
  url: string;
}

export interface swapiPeopleSearch {
  count: number;
  results: swapiPeople[];
}

export interface swapiPlanet {
  name: string;
  rotation_period: string;
  orbital_period: string;
  diameter: string;
  climate: string;
  gravity: string;
  terrain: string;
  surface_water: string;
  population: string;
  residents: string[];
  films: string[];
  created: Date;
  edited: Date;
  url: string;
}

export interface swapiPlanetSearch {
  count: number;
  results: swapiPlanet[];
}
