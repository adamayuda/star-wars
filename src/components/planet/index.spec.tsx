import { swapiPlanet } from "src/types/swapi";
import { Planet } from ".";
import { render } from "@testing-library/react";
import React from "react";

describe("src/components/loading/index.tsx", () => {
  test("Render Loading component", () => {
    const planet: swapiPlanet = {
      name: "Muunilinst",
      rotation_period: "28",
      orbital_period: "412",
      diameter: "13800",
      climate: "temperate",
      gravity: "1",
      terrain: "plains, forests, hills, mountains",
      surface_water: "25",
      population: "5000000000",
      residents: ["http://swapi.dev/api/people/77/"],
      films: [],
      created: new Date("2014-12-20T17:57:47.420000Z"),
      edited: new Date("2014-12-20T20:58:18.519000Z"),
      url: "http://swapi.dev/api/planets/57/",
    };
    const renderedLoading = render(
      <Planet planet={planet} totalPopulation={1} />,
    );
    expect(renderedLoading.container).toMatchSnapshot();
  });
});
