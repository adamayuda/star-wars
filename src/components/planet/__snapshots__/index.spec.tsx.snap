// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/loading/index.tsx Render Loading component 1`] = `
<div>
  <div
    class="planet"
  >
    <h2>
      Muunilinst
    </h2>
    <div
      class="bar"
      style="width: 500000000000%;"
    />
    <p>
      500000000000.000
      % of the total population
    </p>
  </div>
</div>
`;
